src/counts.jl

```diff
@@ -231,22 +231,30 @@ end
 
 
 """
-    addcounts!(dict, x[, wv])
+    addcounts!(dict, x[, wv]; alg = :auto)
 
 Add counts based on `x` to a count map. New entries will be added if new values come up.
 If a weighting vector `wv` is specified, the sum of the weights is used rather than the
 raw counts.
+
+alg can be one of
+    :auto       -   default; uses :radixsort if safe to do so
+    :radixsort  -   if radixsort_safe(eltype(x)) == true then use radixsort to
```

Comment:
`:radixsort` is equivalent to `:auto` in practice, which is weird. Better raise an error if `:radixsort` is passed but the type is unsafe.